### PR TITLE
feat-sb-collection

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

Adds new collection and response model for Storage Bucket

## Test Plan

N/A

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/config/collections.php [hunk 1]
```diff
@@ -1436,6 +1436,87 @@
...
+                'key' => 'encrypted',
+                'type' => Database::SYSTEM_VAR_TYPE_BOOLEAN,
+                'default' => true,
+                'required' => true,
+                'array' => false,
+            ],
+            [
+                '$collection' => Database::SYSTEM_COLLECTION_RULES,
+                'label' => 'Virus Scan',
+                'key' => 'antivirus',
```
Comment (eldadfux) [inline app/config/collections.php:1512]:
```suggestion
                'key' => 'antiVirus',
```

File: app/config/collections.php [hunk 2]
```diff
@@ -1436,6 +1436,87 @@
...
+                'key' => 'allowedFileExtensions',
+                'type' => Database::SYSTEM_VAR_TYPE_TEXT,
+                'default' => '',
+                'required' => true,
+                'array' => false,
+            ],
+            [
+                '$collection' => Database::SYSTEM_COLLECTION_RULES,
+                'label' => 'Encryption',
+                'key' => 'encrypted',
```
Comment (eldadfux) [inline app/config/collections.php:1503]:
```suggestion
                'key' => 'encryption',
```

File: app/config/collections.php [hunk 3]
```diff
@@ -1436,6 +1436,87 @@
...
+                'array' => false,
+            ],
+            [
+                '$collection' => Database::SYSTEM_COLLECTION_RULES,
+                'label' => 'Adapter',
+                'key' => 'adapter',
+                'type' => Database::SYSTEM_VAR_TYPE_TEXT,
+                'default' => '',
+                'required' => true,
+                'array' => false,
```
Comment (eldadfux) [inline app/config/collections.php:1480]: We should probably also add an attribute called `adapterCredentials` for future usage. This attribute can use the DB built in JSON filter for easier internal usage.

File: src/Appwrite/Utopia/Response.php [hunk 4]
```diff
@@ -78,7 +78,8 @@ class Response extends SwooleResponse
     // Storage
     const MODEL_FILE = 'file';
     const MODEL_FILE_LIST = 'fileList';
-    const MODEL_BUCKET = 'bucket'; // - Missing
+    const MODEL_BUCKET = 'bucket';
+    const MODEL_BUCKET_LIST = 'bucketList';
```
Comment (eldadfux) [inline src/Appwrite/Utopia/Response.php:82]: We should also add:

```php
->setModel(new BaseList('Buckets List', self::MODEL_BUCKET_LIST, 'buckets', self::MODEL_BUCKET))
```

File: src/Appwrite/Utopia/Response/Model/Bucket.php [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            'default' => 0,
+            'example' => 100,
+        ])
+        ->addRule('allowedFileExtensions', [
+            'type' => self::TYPE_STRING,
+            'description' => 'Allowed file extensions.',
+            'default' => '',
+            'example' => 'jpg,png',
+        ])
+        ->addRule('encrypted', [
```
Comment (eldadfux) [inline src/Appwrite/Utopia/Response/Model/Bucket.php:63]:
```suggestion
        ->addRule('encryption', [
```

File: src/Appwrite/Utopia/Response/Model/Bucket.php [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            'type' => self::TYPE_BOOLEAN,
+            'description' => 'Bucket is encrypted.',
+            'default' => true,
+            'example' => false,
+        ])
+        ->addRule('antivirus', [
```
Comment (eldadfux) [inline src/Appwrite/Utopia/Response/Model/Bucket.php:69]:
```suggestion
        ->addRule('antiVirus', [
```

**Review by eldadfux — CHANGES_REQUESTED** [review]

File: app/config/collections.php (hunk 3 above)
Comment (lohanidamodar) [reply app/config/collections.php:1480]: added

File: src/Appwrite/Utopia/Response.php (hunk 4 above)
Comment (lohanidamodar) [reply src/Appwrite/Utopia/Response.php:82]: done

**Review by eldadfux — APPROVED** [review]

**Pull request merged**